include/deal.II/differentiation/ad/ad_drivers.h

```diff
@@ -275,6 +275,39 @@ namespace Differentiation
       void
       activate_tape(const typename Types<ADNumberType>::tape_index tape_index);
 
+      /**
+       * Return a flag that, when <code>true</code>, indicates that the retaping
+       * of the dependent function for the chosen @p tape_index is necessary for
+       * a reliable computation to be performed.
+       * This may be necessary a sign comparison within branched operations
+       * yields different results to those computed at the original tape
+       * evaluation point.
+       *
+       * @note The chosen tape index must be greater than
+       * Numbers<ADNumberType>::invalid_tape_index and less than
+       * Numbers<ADNumberType>::max_tape_index.
+       */
+      bool
+      requires_retaping(
+        const typename Types<ADNumberType>::tape_index tape_index) const;
+
+      /**
+       * Return a flag that, when <code>true</code>, indicates that the retaping
+       * of the dependent function is necessary for a reliable computation to be
+       * performed on the active tape.
+       * This may be necessary a sign comparison within branched operations
+       * yields different results to those computed at the original tape
+       * evaluation point.
+       */
+      bool
+      last_action_requires_retaping() const;
```

Comment:
New feature -> Documentation requires review.

Reply:
The documentation of this function appears identical to me to the one of the previous function. How do they differ?